iceoryx_examples/icediscovery_in_c/README.md

```diff
@@ -0,0 +1,134 @@
+# icediscovery in C
+
+## Introduction
+
+This example demonstrates how to search for specific services using iceoryx's
+service discovery. It provides two applications - one offering different
+services and one searching for those with different search queries. The
+behavior and structure is quite similar to the [icediscovery C++ example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery).
+
+<!--## Expected Output-->
+<!-- @todo Add expected output with asciinema recording before v2.0-->
+
+## Code walkthrough
+
+### Offer services
+
+We create several publisher ports which offer their services on construction by
+default. For more dynamism the `cameraPublishers` offer/stop their services
+periodically. If you want more information on how to create publisher ports,
+have a look at the [icedelivery C example](https://github.com/eclipse-iceoryx/iceoryx/tree/v2.0.0/iceoryx_examples/icedelivery_in_c).
+
+### Find services
+
+To be able to search for services, we need to include:
+
+<!--[geoffrey][iceoryx_examples/icediscovery_in_c/iox_c_find_service.c][include service discovery]-->
+```c
+#include "iceoryx_binding_c/service_discovery.h"
+```
+
+To get a handle to the service discovery we call `iox_service_discovery_init`.
+The service discovery requires memory of the amount of
+`sizeof(iox_service_discovery_storage_t)` to be initialized.
```

Comment:
Ok, but this is hidden from the user and an implementation detail. From my point of view the important information is that the user must provide an instance of `iox_service_discovery_storage_t` to an initialization function. The size of objects of this class is not relevant to the user and taking care of by the struct itself.

Technically the user can pass any buffer of sufficient size (requires a cast), but it is not intended to be used that way...

I would hence not mention the detail wit `sizeof`, but not a must change ...